Axinom/mosaic-media-template
# [AB#43917] [AB#42728] ingest error handling improved

# Pull Request

## Prerequisites

      releases)

## Description

- Custom validation will not be executed on ingest document upload, if schema validation fails.
	- Otherwise - custom validation can fail, e.g. because there is an object isntead of an expected array.
- Fallback error messages are now more detailed.
- Mosaic packages updated to make sure errors are mapped better

## Testing Notes

- Try uploading some ingest document, where `images` is an object, e.g. `{ "type": "COVER", "path": "xzy" }`, instead of an array of objects.
	- Expected error: `Ingest Document validation has failed.` with one of the validation errors being `JSON path \"document/items/0/data/images\" should be array,null (line: 36, column: 19)`
- Try uploading some ingrst document, where some invalid genre is specified (one that is not define in the management system), for movie, tvhsow, season, and episode.
	- Expected error: `Metadata update has failed because following genres do not exist: AAction`
	- Localization step will also fail with a message `Unable to start the localization step because the metadata update has failed.` (instead of being stuck in IN_PROGRESS state)
